src/Language/Haskell/LSP/Test.hs

```diff
@@ -229,14 +229,14 @@ runSessionWithHandles' serverProc serverIn serverOut config' caps rootDir sessio
   listenServer serverOut context = do
     msgBytes <- getNextMessage serverOut
 
-    reqMap <- readMVar $ requestMap context
-
-    let msg = fst $ decodeFromServerMsg reqMap msgBytes
-    writeChan (messageChan context) (ServerMessage msg)
+    msg <- modifyMVar (requestMap context) $ \reqMap -> do
+      let (msg, newReqMap) = decodeFromServerMsg reqMap msgBytes
+      writeChan (messageChan context) (ServerMessage msg)
```

Comment:
Ideally this write will should happen after the map has been updated to avoid inconsistency, but I don't think it matters yet (since nothing else consumes that map)